aws-cpp-sdk-core/source/utils/threading/Executor.cpp

```diff
@@ -80,6 +84,7 @@ bool PooledThreadExecutor::SubmitToThread(std::function<void()>&& fn)
         m_tasks.push(fnCpy);
     }
 
+	std::unique_lock<std::mutex> locker(m_syncPointLock);

```

Comment:
notify_one() here could cause false alarm (but harmless), since other thread may immediately fetch the task after line 84 executed. It's better to lock syncPointLock first before m_tasks.push then notify_one()